### ISSUE TITLE
accept True or true in annotations

### ISSUE THREAD

**Issue by tooptoop4** [argoproj/argo-cd#17759]
I mistakenly had below with a True (instead of true) and took a while to realise it did not take effect because the T in True was capitalized

```
metadata:
  annotations:
    argocd.argoproj.io/sync-options: ServerSideApply=True
```

can either an error be thrown if the setting is invalid or the True and true be treated the same? I assume other annotations like Replace=True would also be affected

https://github.com/argoproj/argo-cd/blob/496ac5e2943d1cc7768a1680ec70a5014674300f/controller/state.go#L674

**Comment by jannfis** [comment]
Yeah, I also think the sync option parser could be a little more sophisticated :)

**Comment by KR-Ravindra** [comment]
Adding to this since the same gap applies well beyond the annotation case. On `master` (5c03ec6d) nothing validates sync options anywhere:

- `spec.syncPolicy.syncOptions` is `[]string` with no kubebuilder markers, so the CRD schema is `items: {type: string}` (`pkg/apis/application/v1alpha1/types.go` `SyncOptions`, `AddOption`/`HasOption` are plain `slices.Contains`).
- Every consumer does an exact string match (`controller/sync.go` `HasOption("ApplyOutOfSyncOnly=true")` etc.; `gitops-engine/pkg/sync/sync_context.go` `isPruningDisabled` compares to `"false"` exactly), so `Prune=False`, `ServerSideAply=true` or `Timeout=60` are stored, shown in the UI, and silently do nothing.
- `argocd app set --sync-option` (`cmd/util/app.go`) appends whatever was typed; `ValidatePermissions`/`ValidateRepo` in `util/argo/argo.go` never look at `SyncPolicy`.
- #26032 adds CEL enum validation, but only for the opt-in `v1beta1` Application and only for the spec field; `v1alpha1`, the CLI and the per-resource annotation stay unchecked, and the controller needs the option table in Go anyway.

The valid set already exists (`gitops-engine/pkg/sync/common/types.go` constants plus the Argo CD-only literals `CreateNamespace`, `PrunePropagationPolicy`, `RespectIgnoreDifferences`, `FailOnSharedResource`). A small, cluster-free fix would be one key-to-allowed-values table with `SyncOptions.Validate()`, called from `ValidatePermissions` (as an `InvalidSpecError` or a new warning-type condition), from the CLI `sync-option` flag (hard error listing valid options), and against the annotation in the sync context (resource-level message). Table-driven tests in `types_test.go`, `util/argo/argo_test.go` and `cmd/util/app_test.go`.

Happy to send that PR if maintainers agree a condition (rather than an admission error) is the right severity for existing Applications that may already carry stray strings.

**Comment by crenshaw-dev** [comment]
I think a warning condition on validation failure would indeed be helpful.
